@nrwl/workspace ESLint fails on JS libraries

## Current Behavior
Nx libraries generated with the `--js` flag fail linting.

> _NOTE: This specifically affects libs generated with the default `unitTestRunner: "jest"`._

## Expected Behavior
Freshly generated libraries should pass ESLint.

I'm unsure when this regression was introduced because I primarily use TypeScript libraries.

## Steps to Reproduce
1. `nx g @nrwl/workspace:lib --js --name test-js-lib`
2. `nx lint test-js-lib`
3. ESLint fails on the boilerplate Jest file

## Proposed Solution
This can easily be solved by updating either the root `.eslintrc.json` or project-specific lint configs with the following…
```json
{
  "files": ["*.spec.js", "**/*.test.ts", "**/*.spec.ts"],
  "env": {
    "jest": true
  },
  "rules": {}
}
```

### Failure Logs
```bash
Linting "test-js-lib"...

<PROJECT_ROOT>/libs/test-js-lib/src/lib/test-js-lib.spec.js
  2:1  error  'describe' is not defined  no-undef
  3:3  error  'it' is not defined        no-undef
  4:5  error  'expect' is not defined    no-undef

✖ 3 problems (3 errors, 0 warnings)
```

### Environment
```shell
   Node : 18.3.0
   OS   : darwin x64
   yarn : 1.22.18
   
   nx : 14.4.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.4.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.4.3
   @nrwl/eslint-plugin-nx : 14.4.3
   @nrwl/express : Not Found
   @nrwl/jest : 14.4.3
   @nrwl/js : 14.4.3
   @nrwl/linter : 14.4.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.4.3
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.4.3
   @nrwl/web : 14.4.3
   @nrwl/workspace : 14.4.3
   typescript : 4.7.2
   ---------------------------------------
   Community plugins:
```

--- a/packages/angular/src/generators/add-linting/add-linting.ts
+++ b/packages/angular/src/generators/add-linting/add-linting.ts
@@ -12,6 +12,7 @@ export async function addLintingGenerator(
 ): Promise<GeneratorCallback> {
   const installTask = lintInitGenerator(tree, {
     linter: Linter.EsLint,
+    unitTestRunner: options.unitTestRunner,
     skipPackageJson: options.skipPackageJson,
   });
 

--- a/packages/angular/src/generators/add-linting/schema.d.ts
+++ b/packages/angular/src/generators/add-linting/schema.d.ts
@@ -5,4 +5,5 @@ export interface AddLintingGeneratorSchema {
   setParserOptionsProject?: boolean;
   skipFormat?: boolean;
   skipPackageJson?: boolean;
+  unitTestRunner?: string;
 }

--- a/packages/angular/src/generators/application/lib/add-e2e.ts
+++ b/packages/angular/src/generators/application/lib/add-e2e.ts
@@ -48,6 +48,7 @@ export async function addE2e(tree: Tree, options: NormalizedSchema) {
         eslintFilePatterns: [
           joinPathFragments(options.e2eProjectRoot, '**/*.ts'),
         ],
+        unitTestRunner: options.unitTestRunner,
         skipFormat: true,
         setParserOptionsProject: options.setParserOptionsProject,
         skipPackageJson: options.skipPackageJson,

--- a/packages/angular/src/generators/application/lib/add-linting.ts
+++ b/packages/angular/src/generators/application/lib/add-linting.ts
@@ -14,5 +14,6 @@ export async function addLinting(host: Tree, options: NormalizedSchema) {
     prefix: options.prefix,
     setParserOptionsProject: options.setParserOptionsProject,
     skipPackageJson: options.skipPackageJson,
+    unitTestRunner: options.unitTestRunner,
   });
 }

--- a/packages/angular/src/generators/library/library.ts
+++ b/packages/angular/src/generators/library/library.ts
@@ -168,6 +168,7 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
     projectName: options.name,
     projectRoot: options.projectRoot,
     prefix: options.prefix,
+    unitTestRunner: options.unitTestRunner,
     setParserOptionsProject: options.setParserOptionsProject,
     skipFormat: true,
   });

--- a/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
+++ b/packages/angular/src/generators/ng-add/migrate-from-angular-cli.ts
@@ -96,7 +96,7 @@ export async function migrateFromAngularCli(
       createRootKarmaConfig(tree);
     }
     if (workspaceCapabilities.eslint) {
-      updateRootEsLintConfig(tree, eslintConfig);
+      updateRootEsLintConfig(tree, eslintConfig, options.unitTestRunner);
       cleanupEsLintPackages(tree);
     }
 

--- a/packages/angular/src/generators/ng-add/utilities/e2e.migrator.ts
+++ b/packages/angular/src/generators/ng-add/utilities/e2e.migrator.ts
@@ -310,6 +310,7 @@ export class E2eMigrator extends ProjectMigrator<SupportedTargets> {
         project: this.project.name,
         linter: Linter.EsLint,
         eslintFilePatterns: [`${this.project.newRoot}/**/*.{js,ts}`],
+        unitTestRunner: this.options.unitTestRunner,
         tsConfigPaths: [
           joinPathFragments(this.project.newRoot, 'tsconfig.json'),
         ],

--- a/packages/angular/src/generators/ng-add/utilities/workspace.ts
+++ b/packages/angular/src/generators/ng-add/utilities/workspace.ts
@@ -163,7 +163,8 @@ export function updatePackageJson(tree: Tree): void {
 
 export function updateRootEsLintConfig(
   tree: Tree,
-  existingEsLintConfig: any | undefined
+  existingEsLintConfig: any | undefined,
+  unitTestRunner?: string
 ): void {
   if (tree.exists('.eslintrc.json')) {
     /**
@@ -175,7 +176,7 @@ export function updateRootEsLintConfig(
     tree.delete('.eslintrc.json');
   }
 
-  lintInitGenerator(tree, { linter: Linter.EsLint });
+  lintInitGenerator(tree, { linter: Linter.EsLint, unitTestRunner });
 
   if (!existingEsLintConfig) {
     // There was no eslint config in the root, so we keep the generated one as-is.

--- a/packages/cypress/src/generators/cypress-project/cypress-project.ts
+++ b/packages/cypress/src/generators/cypress-project/cypress-project.ts
@@ -110,7 +110,7 @@ function addProject(tree: Tree, options: CypressProjectSchema) {
 
     if (!project.targets) {
       logger.warn(stripIndents`
-      NOTE: Project, "${options.project}", does not have any targets defined and a baseUrl was not provided. Nx will use  
+      NOTE: Project, "${options.project}", does not have any targets defined and a baseUrl was not provided. Nx will use
       "${options.project}:serve" as the devServerTarget. But you may need to define this target within the project, "${options.project}".
       `);
     }

--- a/packages/js/src/generators/library/library.ts
+++ b/packages/js/src/generators/library/library.ts
@@ -151,8 +151,6 @@ export function addLint(
   return lintProjectGenerator(tree, {
     project: options.name,
     linter: options.linter,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     skipFormat: true,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),

--- a/packages/js/src/generators/library/library.ts
+++ b/packages/js/src/generators/library/library.ts
@@ -151,6 +151,8 @@ export function addLint(
   return lintProjectGenerator(tree, {
     project: options.name,
     linter: options.linter,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     skipFormat: true,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),

--- a/packages/js/src/generators/library/library.ts
+++ b/packages/js/src/generators/library/library.ts
@@ -155,6 +155,7 @@ export function addLint(
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [
       `${options.projectRoot}/**/*.${options.js ? 'js' : 'ts'}`,
     ],

--- a/packages/linter/src/generators/init/init.ts
+++ b/packages/linter/src/generators/init/init.ts
@@ -16,9 +16,11 @@ import {
 
 import { Linter } from '../utils/linter';
 import { containsEslint } from '../utils/eslint-file';
+import { ESLint } from 'eslint';
 
 export interface LinterInitOptions {
   linter?: Linter;
+  unitTestRunner?: string;
   skipPackageJson?: boolean;
 }
 
@@ -86,66 +88,78 @@ const globalTsLintConfiguration = {
   },
 };
 
-const globalEsLintConfiguration = {
-  root: true,
-  ignorePatterns: ['**/*'],
-  plugins: ['@nrwl/nx'],
-  /**
-   * We leverage ESLint's "overrides" capability so that we can set up a root config which will support
-   * all permutations of Nx workspaces across all frameworks, libraries and tools.
-   *
-   * The key point is that we need entirely different ESLint config to apply to different types of files,
-   * but we still want to share common config where possible.
-   */
-  overrides: [
+const getGlobalEsLintConfiguration = (unitTestRunner?: string) => {
+  const config: ESLint.ConfigData = {
+    root: true,
+    ignorePatterns: ['**/*'],
+    plugins: ['@nrwl/nx'],
     /**
-     * This configuration is intended to apply to all "source code" (but not
-     * markup like HTML, or other custom file types like GraphQL)
+     * We leverage ESLint's "overrides" capability so that we can set up a root config which will support
+     * all permutations of Nx workspaces across all frameworks, libraries and tools.
+     *
+     * The key point is that we need entirely different ESLint config to apply to different types of files,
+     * but we still want to share common config where possible.
      */
-    {
-      files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
-      rules: {
-        '@nrwl/nx/enforce-module-boundaries': [
-          'error',
-          {
-            enforceBuildableLibDependency: true,
-            allow: [],
-            depConstraints: [
-              { sourceTag: '*', onlyDependOnLibsWithTags: ['*'] },
-            ],
-          },
-        ],
+    overrides: [
+      /**
+       * This configuration is intended to apply to all "source code" (but not
+       * markup like HTML, or other custom file types like GraphQL)
+       */
+      {
+        files: ['*.ts', '*.tsx', '*.js', '*.jsx'],
+        rules: {
+          '@nrwl/nx/enforce-module-boundaries': [
+            'error',
+            {
+              enforceBuildableLibDependency: true,
+              allow: [],
+              depConstraints: [
+                { sourceTag: '*', onlyDependOnLibsWithTags: ['*'] },
+              ],
+            },
+          ],
+        },
       },
-    },
 
-    /**
-     * This configuration is intended to apply to all TypeScript source files.
-     * See the eslint-plugin-nx package for what is in the referenced shareable config.
-     */
-    {
-      files: ['*.ts', '*.tsx'],
-      extends: ['plugin:@nrwl/nx/typescript'],
       /**
-       * Having an empty rules object present makes it more obvious to the user where they would
-       * extend things from if they needed to
+       * This configuration is intended to apply to all TypeScript source files.
+       * See the eslint-plugin-nx package for what is in the referenced shareable config.
        */
-      rules: {},
-    },
+      {
+        files: ['*.ts', '*.tsx'],
+        extends: ['plugin:@nrwl/nx/typescript'],
+        /**
+         * Having an empty rules object present makes it more obvious to the user where they would
+         * extend things from if they needed to
+         */
+        rules: {},
+      },
 
-    /**
-     * This configuration is intended to apply to all JavaScript source files.
-     * See the eslint-plugin-nx package for what is in the referenced shareable config.
-     */
-    {
-      files: ['*.js', '*.jsx'],
-      extends: ['plugin:@nrwl/nx/javascript'],
       /**
-       * Having an empty rules object present makes it more obvious to the user where they would
-       * extend things from if they needed to
+       * This configuration is intended to apply to all JavaScript source files.
+       * See the eslint-plugin-nx package for what is in the referenced shareable config.
        */
+      {
+        files: ['*.js', '*.jsx'],
+        extends: ['plugin:@nrwl/nx/javascript'],
+        /**
+         * Having an empty rules object present makes it more obvious to the user where they would
+         * extend things from if they needed to
+         */
+        rules: {},
+      },
+    ],
+  };
+  if (unitTestRunner === 'jest') {
+    config.overrides.push({
+      files: ['*.spec.ts', '*.spec.tsx', '*.spec.js', '*.spec.jsx'],
+      env: {
+        jest: true,
+      },
       rules: {},
-    },
-  ],
+    });
+  }
+  return config;
 };
 
 function initTsLint(tree: Tree, options: LinterInitOptions): GeneratorCallback {
@@ -175,7 +189,11 @@ function initEsLint(tree: Tree, options: LinterInitOptions): GeneratorCallback {
     removeDependenciesFromPackageJson(tree, ['@nrwl/linter'], []);
   }
 
-  writeJson(tree, '.eslintrc.json', globalEsLintConfiguration);
+  writeJson(
+    tree,
+    '.eslintrc.json',
+    getGlobalEsLintConfiguration(options.unitTestRunner)
+  );
 
   if (tree.exists('.vscode/extensions.json')) {
     updateJson(tree, '.vscode/extensions.json', (json) => {

--- a/packages/linter/src/generators/lint-project/lint-project.ts
+++ b/packages/linter/src/generators/lint-project/lint-project.ts
@@ -20,6 +20,7 @@ interface LintProjectOptions {
   skipFormat: boolean;
   setParserOptionsProject?: boolean;
   skipPackageJson?: boolean;
+  unitTestRunner?: string;
 }
 
 function createTsLintConfiguration(
@@ -90,6 +91,7 @@ export async function lintProjectGenerator(
 ) {
   const installTask = lintInitGenerator(tree, {
     linter: options.linter,
+    unitTestRunner: options.unitTestRunner,
     skipPackageJson: options.skipPackageJson,
   });
   const projectConfig = readProjectConfiguration(tree, options.project);

--- a/packages/linter/src/generators/lint-project/lint-project.ts
+++ b/packages/linter/src/generators/lint-project/lint-project.ts
@@ -11,6 +11,7 @@ import { Linter } from '../utils/linter';
 import { findEslintFile } from '../utils/eslint-file';
 import { join } from 'path';
 import { lintInitGenerator } from '../init/init';
+import { ESLint } from 'eslint';
 
 interface LintProjectOptions {
   project: string;
@@ -20,6 +21,8 @@ interface LintProjectOptions {
   skipFormat: boolean;
   setParserOptionsProject?: boolean;
   skipPackageJson?: boolean;
+  js?: boolean;
+  unitTestRunner?: string;
 }
 
 function createTsLintConfiguration(
@@ -39,9 +42,9 @@ function createTsLintConfiguration(
 function createEsLintConfiguration(
   tree: Tree,
   projectConfig: ProjectConfiguration,
-  setParserOptionsProject: boolean
+  options: LintProjectOptions
 ) {
-  writeJson(tree, join(projectConfig.root, `.eslintrc.json`), {
+  const eslintJSON: ESLint.ConfigData = {
     extends: [`${offsetFromRoot(projectConfig.root)}${findEslintFile(tree)}`],
     // Include project files to be linted since the global one excludes all files.
     ignorePatterns: ['!**/*'],
@@ -61,7 +64,7 @@ function createEsLintConfiguration(
          * parserOptions.project), the executor will attempt to look for the particular error typescript-eslint gives you
          * and provide feedback to the user.
          */
-        parserOptions: !setParserOptionsProject
+        parserOptions: !options.setParserOptionsProject
           ? undefined
           : {
               project: [`${projectConfig.root}/tsconfig.*?.json`],
@@ -81,7 +84,15 @@ function createEsLintConfiguration(
         rules: {},
       },
     ],
-  });
+  };
+  if (options.js && options.unitTestRunner === 'jest') {
+    eslintJSON.overrides.push({
+      files: ['*.spec.js', '*.spec.jsx', '**/*.test.js', '**/*.test.jsx'],
+      env: { jest: true },
+      rules: {},
+    });
+  }
+  writeJson(tree, join(projectConfig.root, `.eslintrc.json`), eslintJSON);
 }
 
 export async function lintProjectGenerator(
@@ -102,11 +113,7 @@ export async function lintProjectGenerator(
         lintFilePatterns: options.eslintFilePatterns,
       },
     };
-    createEsLintConfiguration(
-      tree,
-      projectConfig,
-      options.setParserOptionsProject
-    );
+    createEsLintConfiguration(tree, projectConfig, options);
   } else {
     projectConfig.targets['lint'] = {
       executor: '@angular-devkit/build-angular:tslint',

--- a/packages/linter/src/generators/lint-project/lint-project.ts
+++ b/packages/linter/src/generators/lint-project/lint-project.ts
@@ -11,7 +11,6 @@ import { Linter } from '../utils/linter';
 import { findEslintFile } from '../utils/eslint-file';
 import { join } from 'path';
 import { lintInitGenerator } from '../init/init';
-import { ESLint } from 'eslint';
 
 interface LintProjectOptions {
   project: string;
@@ -21,8 +20,6 @@ interface LintProjectOptions {
   skipFormat: boolean;
   setParserOptionsProject?: boolean;
   skipPackageJson?: boolean;
-  js?: boolean;
-  unitTestRunner?: string;
 }
 
 function createTsLintConfiguration(
@@ -42,9 +39,9 @@ function createTsLintConfiguration(
 function createEsLintConfiguration(
   tree: Tree,
   projectConfig: ProjectConfiguration,
-  options: LintProjectOptions
+  setParserOptionsProject: boolean
 ) {
-  const eslintJSON: ESLint.ConfigData = {
+  writeJson(tree, join(projectConfig.root, `.eslintrc.json`), {
     extends: [`${offsetFromRoot(projectConfig.root)}${findEslintFile(tree)}`],
     // Include project files to be linted since the global one excludes all files.
     ignorePatterns: ['!**/*'],
@@ -64,7 +61,7 @@ function createEsLintConfiguration(
          * parserOptions.project), the executor will attempt to look for the particular error typescript-eslint gives you
          * and provide feedback to the user.
          */
-        parserOptions: !options.setParserOptionsProject
+        parserOptions: !setParserOptionsProject
           ? undefined
           : {
               project: [`${projectConfig.root}/tsconfig.*?.json`],
@@ -84,15 +81,7 @@ function createEsLintConfiguration(
         rules: {},
       },
     ],
-  };
-  if (options.js && options.unitTestRunner === 'jest') {
-    eslintJSON.overrides.push({
-      files: ['*.spec.js', '*.spec.jsx', '**/*.test.js', '**/*.test.jsx'],
-      env: { jest: true },
-      rules: {},
-    });
-  }
-  writeJson(tree, join(projectConfig.root, `.eslintrc.json`), eslintJSON);
+  });
 }
 
 export async function lintProjectGenerator(
@@ -113,7 +102,11 @@ export async function lintProjectGenerator(
         lintFilePatterns: options.eslintFilePatterns,
       },
     };
-    createEsLintConfiguration(tree, projectConfig, options);
+    createEsLintConfiguration(
+      tree,
+      projectConfig,
+      options.setParserOptionsProject
+    );
   } else {
     projectConfig.targets['lint'] = {
       executor: '@angular-devkit/build-angular:tslint',

--- a/packages/next/src/generators/application/lib/add-linting.ts
+++ b/packages/next/src/generators/application/lib/add-linting.ts
@@ -20,6 +20,7 @@ export async function addLinting(
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [`${options.appProjectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });

--- a/packages/next/src/generators/application/lib/add-linting.ts
+++ b/packages/next/src/generators/application/lib/add-linting.ts
@@ -17,8 +17,6 @@ export async function addLinting(
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.projectName,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/next/src/generators/application/lib/add-linting.ts
+++ b/packages/next/src/generators/application/lib/add-linting.ts
@@ -17,6 +17,8 @@ export async function addLinting(
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.projectName,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/node/src/generators/application/application.ts
+++ b/packages/node/src/generators/application/application.ts
@@ -181,6 +181,8 @@ export async function addLintingToApplication(
   const lintTask = await lintProjectGenerator(tree, {
     linter: options.linter,
     project: options.name,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/node/src/generators/application/application.ts
+++ b/packages/node/src/generators/application/application.ts
@@ -187,6 +187,7 @@ export async function addLintingToApplication(
     eslintFilePatterns: [
       `${options.appProjectRoot}/**/*.${options.js ? 'js' : 'ts'}`,
     ],
+    unitTestRunner: options.unitTestRunner,
     skipFormat: true,
     setParserOptionsProject: options.setParserOptionsProject,
   });

--- a/packages/node/src/generators/application/application.ts
+++ b/packages/node/src/generators/application/application.ts
@@ -181,8 +181,6 @@ export async function addLintingToApplication(
   const lintTask = await lintProjectGenerator(tree, {
     linter: options.linter,
     project: options.name,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/react-native/src/generators/application/application.ts
+++ b/packages/react-native/src/generators/application/application.ts
@@ -9,8 +9,8 @@ import {
   convertNxGenerator,
   Tree,
   formatFiles,
-  joinPathFragments,
   GeneratorCallback,
+  joinPathFragments,
 } from '@nrwl/devkit';
 import { normalizeOptions } from './lib/normalize-options';
 import initGenerator from '../init/init';
@@ -29,14 +29,14 @@ export async function reactNativeApplicationGenerator(
   addProject(host, options);
 
   const initTask = await initGenerator(host, { ...options, skipFormat: true });
-  const lintTask = await addLinting(
-    host,
-    options.projectName,
-    options.appProjectRoot,
-    [joinPathFragments(options.appProjectRoot, 'tsconfig.app.json')],
-    options.linter,
-    options.setParserOptionsProject
-  );
+  const lintTask = await addLinting(host, {
+    ...options,
+    projectRoot: options.appProjectRoot,
+    tsConfigPaths: [
+      joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
+    ],
+  });
+
   const jestTask = await addJest(
     host,
     options.unitTestRunner,

--- a/packages/react-native/src/generators/application/application.ts
+++ b/packages/react-native/src/generators/application/application.ts
@@ -9,7 +9,6 @@ import {
   convertNxGenerator,
   Tree,
   formatFiles,
-  joinPathFragments,
   GeneratorCallback,
 } from '@nrwl/devkit';
 import { normalizeOptions } from './lib/normalize-options';
@@ -32,10 +31,9 @@ export async function reactNativeApplicationGenerator(
   const lintTask = await addLinting(
     host,
     options.projectName,
+    'tsconfig.app.json',
     options.appProjectRoot,
-    [joinPathFragments(options.appProjectRoot, 'tsconfig.app.json')],
-    options.linter,
-    options.setParserOptionsProject
+    options
   );
   const jestTask = await addJest(
     host,

--- a/packages/react-native/src/generators/application/application.ts
+++ b/packages/react-native/src/generators/application/application.ts
@@ -9,6 +9,7 @@ import {
   convertNxGenerator,
   Tree,
   formatFiles,
+  joinPathFragments,
   GeneratorCallback,
 } from '@nrwl/devkit';
 import { normalizeOptions } from './lib/normalize-options';
@@ -31,9 +32,10 @@ export async function reactNativeApplicationGenerator(
   const lintTask = await addLinting(
     host,
     options.projectName,
-    'tsconfig.app.json',
     options.appProjectRoot,
-    options
+    [joinPathFragments(options.appProjectRoot, 'tsconfig.app.json')],
+    options.linter,
+    options.setParserOptionsProject
   );
   const jestTask = await addJest(
     host,

--- a/packages/react-native/src/generators/library/library.ts
+++ b/packages/react-native/src/generators/library/library.ts
@@ -44,14 +44,13 @@ export async function reactNativeLibraryGenerator(
     e2eTestRunner: 'none',
   });
 
-  const lintTask = await addLinting(
-    host,
-    options.name,
-    options.projectRoot,
-    [joinPathFragments(options.projectRoot, 'tsconfig.lib.json')],
-    options.linter,
-    options.setParserOptionsProject
-  );
+  const lintTask = await addLinting(host, {
+    ...options,
+    projectName: options.name,
+    tsConfigPaths: [
+      joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
+    ],
+  });
 
   if (!options.skipTsConfig) {
     updateBaseTsConfig(host, options);

--- a/packages/react-native/src/generators/library/library.ts
+++ b/packages/react-native/src/generators/library/library.ts
@@ -47,9 +47,10 @@ export async function reactNativeLibraryGenerator(
   const lintTask = await addLinting(
     host,
     options.name,
-    'tsconfig.lib.json',
     options.projectRoot,
-    options
+    [joinPathFragments(options.projectRoot, 'tsconfig.lib.json')],
+    options.linter,
+    options.setParserOptionsProject
   );
 
   if (!options.skipTsConfig) {

--- a/packages/react-native/src/generators/library/library.ts
+++ b/packages/react-native/src/generators/library/library.ts
@@ -47,10 +47,9 @@ export async function reactNativeLibraryGenerator(
   const lintTask = await addLinting(
     host,
     options.name,
+    'tsconfig.lib.json',
     options.projectRoot,
-    [joinPathFragments(options.projectRoot, 'tsconfig.lib.json')],
-    options.linter,
-    options.setParserOptionsProject
+    options
   );
 
   if (!options.skipTsConfig) {

--- a/packages/react-native/src/utils/add-linting.spec.ts
+++ b/packages/react-native/src/utils/add-linting.spec.ts
@@ -16,13 +16,12 @@ describe('Add Linting', () => {
   });
 
   it('should add update `workspace.json` file properly when eslint is passed', () => {
-    addLinting(
-      tree,
-      'my-lib',
-      'libs/my-lib',
-      ['libs/my-lib/tsconfig.lib.json'],
-      Linter.EsLint
-    );
+    addLinting(tree, {
+      projectName: 'my-lib',
+      linter: Linter.EsLint,
+      tsConfigPaths: ['libs/my-lib/tsconfig.lib.json'],
+      projectRoot: 'libs/my-lib',
+    });
     const project = readProjectConfiguration(tree, 'my-lib');
 
     expect(project.targets.lint).toBeDefined();
@@ -30,13 +29,12 @@ describe('Add Linting', () => {
   });
 
   it('should add update `workspace.json` file properly when tslint is passed', () => {
-    addLinting(
-      tree,
-      'my-lib',
-      'libs/my-lib',
-      ['libs/my-lib/tsconfig.lib.json'],
-      Linter.TsLint
-    );
+    addLinting(tree, {
+      projectName: 'my-lib',
+      linter: Linter.TsLint,
+      tsConfigPaths: ['libs/my-lib/tsconfig.lib.json'],
+      projectRoot: 'libs/my-lib',
+    });
     const project = readProjectConfiguration(tree, 'my-lib');
 
     expect(project.targets.lint).toBeDefined();
@@ -46,13 +44,12 @@ describe('Add Linting', () => {
   });
 
   it('should not add lint target when "none" is passed', async () => {
-    addLinting(
-      tree,
-      'my-lib',
-      'libs/my-lib',
-      ['libs/my-lib/tsconfig.lib.json'],
-      Linter.None
-    );
+    addLinting(tree, {
+      projectName: 'my-lib',
+      linter: Linter.None,
+      tsConfigPaths: ['libs/my-lib/tsconfig.lib.json'],
+      projectRoot: 'libs/my-lib',
+    });
     const project = readProjectConfiguration(tree, 'my-lib');
 
     expect(project.targets.lint).toBeUndefined();

--- a/packages/react-native/src/utils/add-linting.ts
+++ b/packages/react-native/src/utils/add-linting.ts
@@ -9,38 +9,39 @@ import {
 import { extraEslintDependencies, createReactEslintJson } from '@nrwl/react';
 import type { Linter as ESLintLinter } from 'eslint';
 
-export async function addLinting(
-  host: Tree,
-  projectName: string,
-  appProjectRoot: string,
-  tsConfigPaths: string[],
-  linter: Linter,
-  setParserOptionsProject?: boolean
-) {
-  if (linter === Linter.None) {
+interface NormalizedSchema {
+  linter?: Linter;
+  projectName: string;
+  projectRoot: string;
+  setParserOptionsProject?: boolean;
+  tsConfigPaths: string[];
+}
+
+export async function addLinting(host: Tree, options: NormalizedSchema) {
+  if (options.linter === Linter.None) {
     return () => {};
   }
 
   const lintTask = await lintProjectGenerator(host, {
-    linter,
-    project: projectName,
-    tsConfigPaths,
-    eslintFilePatterns: [`${appProjectRoot}/**/*.{ts,tsx,js,jsx}`],
+    linter: options.linter,
+    project: options.projectName,
+    tsConfigPaths: options.tsConfigPaths,
+    eslintFilePatterns: [`${options.projectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });
 
-  if (linter === Linter.TsLint) {
+  if (options.linter === Linter.TsLint) {
     return () => {};
   }
 
   const reactEslintJson = createReactEslintJson(
-    appProjectRoot,
-    setParserOptionsProject
+    options.projectRoot,
+    options.setParserOptionsProject
   );
 
   updateJson(
     host,
-    joinPathFragments(appProjectRoot, '.eslintrc.json'),
+    joinPathFragments(options.projectRoot, '.eslintrc.json'),
     (json: ESLintLinter.Config) => {
       json = reactEslintJson;
       json.ignorePatterns = ['!**/*', 'public', '.cache', 'node_modules'];

--- a/packages/react-native/src/utils/add-linting.ts
+++ b/packages/react-native/src/utils/add-linting.ts
@@ -9,38 +9,52 @@ import {
 import { extraEslintDependencies, createReactEslintJson } from '@nrwl/react';
 import type { Linter as ESLintLinter } from 'eslint';
 
+interface LintingSchema {
+  linter?: Linter;
+  js?: boolean;
+  unitTestRunner?: 'jest' | 'none';
+  setParserOptionsProject?: boolean;
+}
+
+// options.name,
+// options.projectRoot,
+// [joinPathFragments(options.projectRoot, 'tsconfig.lib.json')],
+// options.linter,
+// options.setParserOptionsProject
+
 export async function addLinting(
   host: Tree,
   projectName: string,
-  appProjectRoot: string,
-  tsConfigPaths: string[],
-  linter: Linter,
-  setParserOptionsProject?: boolean
+  tsConfigFileName: string,
+  projectRoot: string,
+  options: LintingSchema
 ) {
-  if (linter === Linter.None) {
+  if (options.linter === Linter.None) {
     return () => {};
   }
 
   const lintTask = await lintProjectGenerator(host, {
-    linter,
+    linter: options.linter,
     project: projectName,
-    tsConfigPaths,
-    eslintFilePatterns: [`${appProjectRoot}/**/*.{ts,tsx,js,jsx}`],
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
+    tsConfigPaths: [joinPathFragments(projectRoot, tsConfigFileName)],
+    eslintFilePatterns: [`${projectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });
 
-  if (linter === Linter.TsLint) {
+  if (options.linter === Linter.TsLint) {
     return () => {};
   }
 
   const reactEslintJson = createReactEslintJson(
-    appProjectRoot,
-    setParserOptionsProject
+    projectRoot,
+    options.setParserOptionsProject
   );
 
   updateJson(
     host,
-    joinPathFragments(appProjectRoot, '.eslintrc.json'),
+    joinPathFragments(projectRoot, '.eslintrc.json'),
     (json: ESLintLinter.Config) => {
       json = reactEslintJson;
       json.ignorePatterns = ['!**/*', 'public', '.cache', 'node_modules'];

--- a/packages/react-native/src/utils/add-linting.ts
+++ b/packages/react-native/src/utils/add-linting.ts
@@ -9,52 +9,38 @@ import {
 import { extraEslintDependencies, createReactEslintJson } from '@nrwl/react';
 import type { Linter as ESLintLinter } from 'eslint';
 
-interface LintingSchema {
-  linter?: Linter;
-  js?: boolean;
-  unitTestRunner?: 'jest' | 'none';
-  setParserOptionsProject?: boolean;
-}
-
-// options.name,
-// options.projectRoot,
-// [joinPathFragments(options.projectRoot, 'tsconfig.lib.json')],
-// options.linter,
-// options.setParserOptionsProject
-
 export async function addLinting(
   host: Tree,
   projectName: string,
-  tsConfigFileName: string,
-  projectRoot: string,
-  options: LintingSchema
+  appProjectRoot: string,
+  tsConfigPaths: string[],
+  linter: Linter,
+  setParserOptionsProject?: boolean
 ) {
-  if (options.linter === Linter.None) {
+  if (linter === Linter.None) {
     return () => {};
   }
 
   const lintTask = await lintProjectGenerator(host, {
-    linter: options.linter,
+    linter,
     project: projectName,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
-    tsConfigPaths: [joinPathFragments(projectRoot, tsConfigFileName)],
-    eslintFilePatterns: [`${projectRoot}/**/*.{ts,tsx,js,jsx}`],
+    tsConfigPaths,
+    eslintFilePatterns: [`${appProjectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });
 
-  if (options.linter === Linter.TsLint) {
+  if (linter === Linter.TsLint) {
     return () => {};
   }
 
   const reactEslintJson = createReactEslintJson(
-    projectRoot,
-    options.setParserOptionsProject
+    appProjectRoot,
+    setParserOptionsProject
   );
 
   updateJson(
     host,
-    joinPathFragments(projectRoot, '.eslintrc.json'),
+    joinPathFragments(appProjectRoot, '.eslintrc.json'),
     (json: ESLintLinter.Config) => {
       json = reactEslintJson;
       json.ignorePatterns = ['!**/*', 'public', '.cache', 'node_modules'];

--- a/packages/react/src/generators/application/application.ts
+++ b/packages/react/src/generators/application/application.ts
@@ -35,6 +35,7 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [`${options.appProjectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });

--- a/packages/react/src/generators/application/application.ts
+++ b/packages/react/src/generators/application/application.ts
@@ -32,6 +32,8 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.projectName,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/react/src/generators/application/application.ts
+++ b/packages/react/src/generators/application/application.ts
@@ -32,8 +32,6 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.projectName,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],

--- a/packages/react/src/generators/library/library.ts
+++ b/packages/react/src/generators/library/library.ts
@@ -147,6 +147,7 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [`${options.projectRoot}/**/*.{ts,tsx,js,jsx}`],
     skipFormat: true,
   });

--- a/packages/react/src/generators/library/library.ts
+++ b/packages/react/src/generators/library/library.ts
@@ -144,6 +144,8 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.name,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
     ],

--- a/packages/react/src/generators/library/library.ts
+++ b/packages/react/src/generators/library/library.ts
@@ -144,8 +144,6 @@ async function addLinting(host: Tree, options: NormalizedSchema) {
   const lintTask = await lintProjectGenerator(host, {
     linter: options.linter,
     project: options.name,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
     ],

--- a/packages/web/src/generators/application/application.ts
+++ b/packages/web/src/generators/application/application.ts
@@ -206,6 +206,7 @@ export async function applicationGenerator(host: Tree, schema: Schema) {
     tsConfigPaths: [
       joinPathFragments(options.appProjectRoot, 'tsconfig.app.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [`${options.appProjectRoot}/**/*.ts`],
     skipFormat: true,
     setParserOptionsProject: options.setParserOptionsProject,

--- a/packages/workspace/src/generators/library/library.ts
+++ b/packages/workspace/src/generators/library/library.ts
@@ -77,6 +77,8 @@ export function addLint(
   return lintProjectGenerator(tree, {
     project: options.name,
     linter: options.linter,
+    js: options.js,
+    unitTestRunner: options.unitTestRunner,
     skipFormat: true,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),

--- a/packages/workspace/src/generators/library/library.ts
+++ b/packages/workspace/src/generators/library/library.ts
@@ -77,8 +77,6 @@ export function addLint(
   return lintProjectGenerator(tree, {
     project: options.name,
     linter: options.linter,
-    js: options.js,
-    unitTestRunner: options.unitTestRunner,
     skipFormat: true,
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),

--- a/packages/workspace/src/generators/library/library.ts
+++ b/packages/workspace/src/generators/library/library.ts
@@ -81,6 +81,7 @@ export function addLint(
     tsConfigPaths: [
       joinPathFragments(options.projectRoot, 'tsconfig.lib.json'),
     ],
+    unitTestRunner: options.unitTestRunner,
     eslintFilePatterns: [
       `${options.projectRoot}/**/*.${options.js ? 'js' : 'ts'}`,
     ],

